### PR TITLE
Toolbar Example: Set focus to button with tabindex="0" when the container is clicked 

### LINKED ISSUE #987: Toolbar Example: Make clicking padding move focus into toolbar
Per [comment in issue 539](https://github.com/w3c/aria-practices/issues/539#issuecomment-454870090)
by @sh0ji about the [toolbar example](http://w3c.github.io/aria-practices/examples/toolbar/toolbar.html),
make clicking the padding of the toolbar (the gray #ECECEA part) move focus into the toolbar.
This enables people to use their mouse to orient their focus and then use their keyboard to navigate inside the toolbar.

### PR DESCRIPTION
**Pull request by spectranaut**
This PR is an adoption of the work of @maschad96 in #1384. Somehow the work was never reviewed until now, and there are some conflicts and some fixes so I'm opened a new PR for it! Good work @maschad96, especially on the tests!

The original description:

Resolves #987

[Preview Link](https://raw.githack.com/w3c/aria-practices/pr-1384-maschad96git/examples/toolbar/toolbar.html)

The issue mentions that:
- When focus is on a button, a focus ring is shown around the toolbar container.
- As a result, it conveys visually that focus is managed within that larger container

This approach basically adds another prototype method for `handleContainerClick`, comparing `this.domNode` to `event.target` to determine if it should set focus.

Open to feedback and other approaches. Thanks!

### CODE REVIEW

**Comment by mcking65** [comment]
@jongund @smhigley @carmacleod 
Available to review this?

**Review by jongund — APPROVED** [review]
@spectranaut Thanks for implementing the fix

**Comment by mcking65** [comment]
@jongund thank you for the speedy review!

File: test/tests/toolbar_toolbar.js [hunk 1]
```diff
@@ -26,11 +25,20 @@ const ex = {
...
+    second: '#ex1 .item:nth-child(2)',
+    last: '#ex1 #link',
+  },
+  radioButtons: {
+    first: '#ex1 [role="radio"]:nth-child(1)',
+    second: '#ex1 [role="radio"]:nth-child(2)',
+    last: '#ex1 [role="radio"]:nth-child(3)',
+  },
+  tabbaleItemAfterToolbarSelector: '#textarea1',
+  tabbaleItemBeforeToolbarSelector: 'body > main > p > a:last-of-type',
```
Comment (carmacleod) [inline test/tests/toolbar_toolbar.js:40]: Missing 'b' in tabbable:
```suggestion
  tabbableItemAfterToolbarSelector: '#textarea1',
  tabbableItemBeforeToolbarSelector: 'body > main > p > a:last-of-type',
```

File: test/tests/toolbar_toolbar.js [hunk 2]
```diff
@@ -462,169 +449,344 @@ ariaTest('', exampleFile, 'toolbar-spinbutton-aria-valuemax', async (t) => {
...
+    await assertHasFocus(t, ex.itemSelectors.first);
+  }
+);
 
+ariaTest(
+  'TAB: Moves focus into the toolbar, to the first menu item',
+  exampleFile,
+  'toolbar-tab',
+  async (t) => {
+    let tabTarget = ex.tabbaleItemBeforeToolbarSelector;
```
Comment (carmacleod) [inline test/tests/toolbar_toolbar.js:555]: Missing 'b' in tabbable:
```suggestion
    let tabTarget = ex.tabbableItemBeforeToolbarSelector;
```

File: test/tests/toolbar_toolbar.js [hunk 3]
```diff
@@ -462,169 +449,344 @@ ariaTest('', exampleFile, 'toolbar-spinbutton-aria-valuemax', async (t) => {
...
-    let nextToolSelector = ex.allToolSelectors[index + 1];
+ariaTest(
+  'TAB: Moves focus out of the toolbar, to the next control',
+  exampleFile,
+  'toolbar-tab',
+  async (t) => {
+    await t.context.session
+      .findElement(By.css(ex.itemSelector))
+      .sendKeys(Key.TAB);
+    await assertHasFocus(t, ex.tabbaleItemAfterToolbarSelector);
```
Comment (carmacleod) [inline test/tests/toolbar_toolbar.js:569]: Missing 'b' in tabbable:
```suggestion
    await assertHasFocus(t, ex.tabbableItemAfterToolbarSelector);
```

**Review by carmacleod — APPROVED** [review]
+1
Looks good! Nice feature.
Only a few places in the tests where "tabbable" is missing its 3rd 'b'. 😄

**Comment by mcking65** [comment]
@carmacleod thank you for catching those ... are the test files excluded from spell check? It passed.

**Comment by nschonni** [comment]
I think it's the `allowCompoundWords` in https://github.com/w3c/aria-practices/blob/1e11b3d3cf6e16c9212a742dc38868286c838f8e/cspell.json#L213, so it sees it as "tab bale" as 2 possible correct words

**Comment by mcking65** [comment]
@nschonni 
Is there a way to restrict compound word passing only to when they are camel cased?

**Comment by nschonni** [comment]
@mcking65 camel case is already treated as separate words. I just spun up #1702 to keep the discussion separate 😄

**Review by mcking65 — APPROVED** [review]
@maschad96 thank you very much for this contribution! And, @spectranaut thank you for helping bring it up to date.

**Pull request merged**